lib/active_merchant/billing/gateways/nuvei.rb

```diff
@@ -415,12 +415,12 @@ def current_timestamp
         Time.now.utc.strftime('%Y%m%d%H%M%S')
       end
 
-      def build_post_data(post)
+      def build_post_data(post, options = {})
         post[:merchantId] = @options[:merchant_id]
         post[:merchantSiteId] = @options[:merchant_site_id]
         post[:timeStamp] = current_timestamp.to_i
         post[:clientRequestId] = SecureRandom.uuid
-        post[:clientUniqueId] = SecureRandom.hex(16)
+        post[:clientUniqueId] = options[:order_id] if options[:order_id]
```

Comment:
Since this is a required field, we should default to the current SecureRandom output. The Nuvei API [ref](https://docs.nuvei.com/api/main/indexMain_v1_0.html?json#initPayment).